app/models/bulkrax/csv_entry.rb

```diff
@@ -49,12 +51,26 @@ def matcher_class
       Bulkrax::CsvMatcher
     end
 
+    def collections_created?
+      return true if record['collection'].blank?
+      record['collection'].split(/\s*[:;|]\s*/).length == self.collection_ids.length
+    end
+
+    def collections
+      return self.collection_ids if collections_created?
```

Comment:
its still odd to me for a method called `collections` to return an array of ids and not the collections themselves.

Reply:
I've renamed to `retrieve_or_create_collection_ids`